core/src/main/java/org/hisp/dhis/android/core/constant/internal/KtorConstantService.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2004-2024, University of Oslo
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  Redistributions of source code must retain the above copyright notice, this
+ *  list of conditions and the following disclaimer.
+ *
+ *  Redistributions in binary form must reproduce the above copyright notice,
+ *  this list of conditions and the following disclaimer in the documentation
+ *  and/or other materials provided with the distribution.
+ *  Neither the name of the HISP project nor the names of its contributors may
+ *  be used to endorse or promote products derived from this software without
+ *  specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.android.core.constant.internal
+
+import io.ktor.client.request.get
+import org.hisp.dhis.android.core.arch.api.fields.internal.Fields
+import org.hisp.dhis.android.core.arch.api.internal.KtorServiceClient
+import org.hisp.dhis.android.core.arch.api.payload.internal.Payload
+import org.hisp.dhis.android.core.constant.Constant
+import org.koin.core.annotation.Singleton
+
+@Singleton
+internal class KtorConstantService(private val client: KtorServiceClient) {
+    suspend fun constants(fields: Fields<Constant>, paging: Boolean): Payload<Constant> {
+        val url = "${URL}constants"
+        val parameters = mapOf(
+            "fields" to fields.generateString(),
+            "paging" to paging.toString(),
+        )
+        return client.get(url, parameters)
+    }
```

Comment:
Maybe not for now, but I think we can avoid repetition of parameter names all across the services. For example, the parameters `fields` or `paging` are used in almost every method.

Just brainstorming, we could use a DSL to call the method something like:
```
return client.get {
  url("constants")
  parameters {
    fields(fields)
    paging(paging)
  }
}
```
Then, the method in KtorServiceClient will translate those parameters to `fields` and `paging`. And it will call the method `.generateString()` in the fields.

We can talk about that in another moment.